nodes/node_battery_holder.lua

```diff
@@ -67,28 +67,11 @@ local def = {
 	-- Allow all items with energy storage to be placed in the inventory
 	allow_metadata_inventory_put = function(pos, listname, _, stack, player)
 		if listname == "batteries" then
-			if not minetest.global_exists("technic") then
-				return 0
-			end
-
-			local node_name = stack:get_name()
-
-			-- Allow all items with energy storage from technic mod
-			if technic.power_tools[node_name] ~= nil then
-				local meta = stack:get_metadata()
-				local md = minetest.deserialize(meta)
-				-- And specifically if they hold any charge
-				-- Disregard empty batteries, the player should know better
-				if md and md.charge > 0 then
-					if digtron.check_protected_and_record(pos, player) then
-						return 0
-					end
-					return stack:get_count()
-				else
+			if minetest.global_exists("technic") and technic.get_charge(stack) > 0 then
+				if digtron.check_protected_and_record(pos, player) then
 					return 0
 				end
-			else
-				return 0
+				return stack:get_count()
```

Comment:
Do you *really* intend to allow players to use this slot for arbitrary storage if technic is not installed?